src/sentry/event_manager.py

```diff
@@ -329,8 +329,16 @@ def _normalize_impl(self, project_id=None):
             **DEFAULT_STORE_NORMALIZER_ARGS,
         )
 
+        # TODO check if this is a right way to do this
+        # This bypasses transaction normalization to preserve fingerprint
+        fingerprint = self._data.get("fingerprint", None)
+        if fingerprint:
+            del self._data["fingerprint"]
         self._data = CanonicalKeyDict(rust_normalizer.normalize_event(dict(self._data)))
 
```

Comment:
I'm not sure how I can update `rust_normalizer` to accept `fingerprint` on transaction events

Reply:
Damn, I didn't realise we'd need to change how event normalization works here... It might be worth asking ingest team about this. I'm not sure if there's another way we could inject this into the event

Reply:
nevermind, I had a typo when writing a fingerprint to transaction which is why normalizer was failing